### PR TITLE
Docs improvements to #1730

### PR DESCRIPTION
**Pull request by dvagnero**
Expand procedure
Add server config example

#1730

### CODE REVIEW

File: documentation/asciidoc/topics/proc_securing_custom_configuration.adoc [hunk 1]
```diff
@@ -34,8 +34,16 @@ include::yaml/credential_store.yaml[]
 ----
 
 . Apply the changes.
-. Define a `credential-reference` in your configuration with `store="credentials"`.
+. Open your {brandname} Server for editing.
```
Comment (ryanemerson) [inline documentation/asciidoc/topics/proc_securing_custom_configuration.adoc:37]: I think we might need to add "configuration" here :slightly_smiling_face:

File: documentation/asciidoc/topics/proc_securing_custom_configuration.adoc [hunk 2]
```diff
@@ -34,8 +34,16 @@ include::yaml/credential_store.yaml[]
...
+. Add a `credential-reference` in the `security`  section of the configuration.
```
Comment (ryanemerson) [inline documentation/asciidoc/topics/proc_securing_custom_configuration.adoc:38]: This isn't correct. A `credential-reference` can appear in many parts of the configuration, for example in a DataSource connection factory or a Security-realm.

https://infinispan.org/docs/stable/titles/server/server.html#credential-keystore-references_credential-keystores

I think we can skip the `security` section part of this bullet and just say something like "Add a `credential-reference` to your configuration".

File: documentation/asciidoc/topics/proc_securing_custom_configuration.adoc [hunk 3]
```diff
@@ -34,8 +34,16 @@ include::yaml/credential_store.yaml[]
...
+. Specify `credentials` as the name of the `store`.
+. Specify the key of the secret with the `alias` attribute.
```
Comment (ryanemerson) [inline documentation/asciidoc/topics/proc_securing_custom_configuration.adoc:40]: Should these two bullets be nested under the "Add a `credential-reference`..." bullet as they're related?

Also, I think this sentence might be easier to understand if it is flipped, e.g.:

"Specify the `alias` attribute as one of the keys defined in your credential Secret"

**Comment by dvagnero** [comment]
Thanks @ryanemerson for your review and suggestions. I think it looks good now

**Comment by ryanemerson** [comment]
Thanks @dvagnero

**Pull request merged**